src/QuadForm/Quad/ZLattices.jl

```diff
@@ -69,16 +69,10 @@ end
 
 Return the Z-lattice with basis matrix $B$ inside the quadratic space $V$.
 """
-function lattice(V::QuadSpace{FlintRationalField, fmpq_mat}, B::MatElem; isbasis::Bool = true, check::Bool = true)
+function lattice(V::QuadSpace{FlintRationalField, fmpq_mat}, B::Union{fmpz_mat, fmpq_mat}; isbasis::Bool = true, check::Bool = true)
```

Comment:
Seems unnecessarily restrictive to me?

Reply:
Replacing by `MatElem{<: RationalUnion}` would be more reasonable ?

Reply:
Probably both are ok.